sources/models/shared-defaults/kubernetes-aws.toml

```diff
@@ -5,6 +5,17 @@ authentication-mode = "aws"
 server-tls-bootstrap = true
 cloud-provider = "aws"
 
+[settings.kubernetes.credential-providers.ecr-credential-provider]
+enabled = true
+cache-duration = "12h"
+image-patterns = [
+    "*.dkr.ecr.*.amazonaws.com",
+    "*.dkr.ecr.*.amazonaws.cn",
+    "*.dkr.ecr-fips.*.amazonaws.com",
+    "*.dkr.ecr.us-iso-east-1.c2s.ic.gov",
+    "*.dkr.ecr.us-isob-east-1.sc2s.sgov.gov"
+]
```

Comment:
Do we really want to switch to the credential provider by default for all existing k8s variants? That feels like it's more risk than necessary.

Some alternatives:
* create separate `sources/models/src/aws-k8s-1.27` and `sources/models/src/aws-k8s-1.27-nvidia` trees (as we eventually did for 1.26, so we could add `kubernetes-aws-external-cloud-provider.toml`)
* use the credential provider for aws-k8s-1.26 also

I don't love either but the separate model seems like the safest path.

Reply:
Separated.